slam-libraries/viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -143,6 +147,10 @@ class SLAMServiceImpl final : public SLAMService::Service {
 
     std::mutex map_builder_mutex;
     mapping::MapBuilder map_builder;
+
+    std::mutex pose_mutex;
+    cartographer::transform::Rigid3d latest_global_pose =
```

Comment:
[q] why isn't this just global pose? Is there another variable with that name already im not seeing?

Reply:
I think either is fine. `latest_global_pose` feels more accurate since we can not guarantee to know the `global_pose` at all times, only the latest global pose we grabbed & updated. Adding `latest_` therefore just implies that there's a certain amount of uncertainty in this value. But I don't think it matters much either way.

Reply:
yeah latest just kinda stuck. Wanted it to be distinct because we were grabbing a specific point in our trajectory

Reply:
gotcha, was just confused about the switch in local pose to remove latest and why it was kept here

Reply:
oh yeah the function doesn't necessarily need the latest pose which is why we removed it from that